### ISSUE TITLE
Library allows add_peer() when connection_state is not WebRTCPeerConnection.STATE_NEW.

### ISSUE THREAD

**Issue by goatchurchprime** [godotengine/webrtc-native#47]
### Godot version

3.3.3.stable

### Plugin version

0.5

### System information

Linux

### Issue description

According to [the documentation for add_peer](https://docs.godotengine.org/en/stable/classes/class_webrtcmultiplayer.html?#class-webrtcmultiplayer-method-add-peer), the following should not have worked:

```GDscript
if v["subject"] == "offer":
	var peer = WebRTCPeerConnection.new()
	peer.initialize({"iceServers": [ { "urls": ["stun:stun.l.google.com:19302"] } ] })
	peer.connect("session_description_created", self, "client_session_description_created")
	peer.connect("ice_candidate_created", self, "client_ice_candidate_created")

	var networkedmultiplayerclient = WebRTCMultiplayer.new()
	peer.set_remote_description("offer", v["data"])
	networkedmultiplayerclient.add_peer(peer, 1)
```

However, it does in the webrtc-native library.  But it doesn't in the HTML5 implementation.  This difference made my bug quite hard to find.  

Basically my last two lines needed to be swapped around for it to be right.

### Steps to reproduce

The code above is an excerpt of the client_packet_received from here: 
https://github.com/goatchurchprime/godot_multiplayer_networking_workbench/blob/main/networking/WebRTCmultiplayerclient.gd#L29

It would take a bit of work to thin it down to just that WebRTCmultiplayerclient example.  It would be easy to show if there was a minimal WebRTC multiplayer example in the [networking demo projects](https://github.com/godotengine/godot-demo-projects/tree/3.3-f9333dc/networking), and to keep it self-contained it could use WebSockets as the signalling system

### Minimal reproduction project

See above

**Comment by goatchurchprime** [comment]
I've finally got the example derived from the signalling demo with an ugly toggle button "Addpeerbeforesetdescription" to swap the order of the function calls.

[webrtc_signaling (copy).zip](https://github.com/godotengine/webrtc-native/files/7602280/webrtc_signaling.copy.zip)

I've used a `peerslist = {}` to keep track of the peers created, instead of fetching them back using the `WebRTCMultiplayer.get_peer(id).connection` function.  

It's necessary to call `WebRTCPeerConnection.add_peer(peer)` before `peer.create_offer()`

However on the receiving end of the Offer your library lets me get away with `add_peer()` *after* `set_remote_description()`:

```GDscript
  var rtc_mp = WebRTCMultiplayer.new()
  peer.set_remote_description("offer", offer)
  rtc_mp.add_peer(peer, id)
```
I have confirmed that the HTML5 implementation does not work for this, but your library does.  

To run the HTML5 version:
* Run a copy of `WebRTC_signalling_demo` on the command line and click on the Signalling server: `Listen` button.  
* Run the exported HTML5
* Hit `Start` on the top left panel
* Paste the string after `Joined lobby` into the Room `Secret` line edit box
* Hit `Start` on the top right panel
* ... then some stuff happens (or doesn't) to indicate a WebRTC connection has been established.

This is now running on Godot version 3.4.  

The reason this issue matters is it's easier to develop WebRTC code against this plugin than on the HTML5 platform, so this is where people are going to be looking for their bugs.
